x-pack/plugins/security/server/config.ts

```diff
@@ -149,18 +149,20 @@ const providersConfigSchema = schema.object(
         }),
       },
       {
-        credentials: schema.oneOf([
-          schema.object({
-            username: schema.string(),
-            password: schema.string(),
-          }),
-          schema.object({
-            apiKey: schema.oneOf([
-              schema.object({ id: schema.string(), key: schema.string() }),
-              schema.string(),
-            ]),
-          }),
-        ]),
+        credentials: schema.maybe(
+          schema.oneOf([
+            schema.object({
+              username: schema.string(),
+              password: schema.string(),
+            }),
+            schema.object({
+              apiKey: schema.oneOf([
+                schema.object({ id: schema.string(), key: schema.string() }),
+                schema.string(),
+              ]),
+            }),
+          ])
+        ),
```

Comment:
What do you think about making this an explicit setting, rather than implicit? Something like:


```suggestion
        credentials: schema.oneOf([
          schema.literal('elasticsearch_anonymous_user'),
          schema.object({
            username: schema.string(),
            password: schema.string(),
          }),
          schema.object({
            apiKey: schema.oneOf([
              schema.object({ id: schema.string(), key: schema.string() }),
              schema.string(),
            ]),
          }),
        ]),
```

Reply:
I initially thought that it'd be unnecessary, but the more I think about it the more I agree with you that being explicit in this area makes more sense. And I'll make sure to make it clear in the docs that it's just an arbitrary string and should not be confused with `xpack.security.authc.anonymous.username` in `elasticsearch.yml`.